storage-api/src/java/org/apache/hadoop/hive/common/io/encoded/EncodedColumnBatch.java

```diff
@@ -78,13 +79,21 @@ public void setIndexBaseOffset(int indexBaseOffset) {
 
     @Override
     public String toString() {
-      String bufStr = "";
+      StringBuilder sb = new StringBuilder();
       if (cacheBuffers != null) {
-        for (MemoryBuffer mb : cacheBuffers) {
-          bufStr += mb.getClass().getSimpleName() + " with " + mb.getByteBufferRaw().remaining() + " bytes, ";
+        Iterator<MemoryBuffer> iter = cacheBuffers.iterator();
+        while (iter.hasNext()) {
+          MemoryBuffer mb = iter.next();
+          sb.append(mb.getClass().getSimpleName());
+          sb.append(" with ");
+          sb.append(mb.getByteBufferRaw().remaining());
+          sb.append(" bytes");
+          if (iter.hasNext()) {
+            sb.append(", ");
```

Comment:
Will have to check this.  I think from my own patch experience, that this change might break a bunch of unit tests because they are checking the output of `toString()` which previously always adds a comma, even if it's not really needed.  I'm not 100%, but just FYI.

Reply:
Tests passed so this change should be safe.